cmd/find.go

```diff
@@ -316,89 +342,33 @@ func stringsReplace(args string, fileContent contentMessage) string {
 	return str
 }
 
-// watchEventsFind used in conjunction with doFindWatch method to detect
-// and preform desired action when an object is created
-func watchEventsFind(ctx *cli.Context, clnt Client, params watchParams, alias string, fileContent contentMessage) {
-	watchObj, err := clnt.Watch(params)
-	fatalIf(err.Trace(alias), "Cannot watch with given params")
-
-	// enables users to kill using the control + c
-	trapCh := signalTrap(os.Interrupt, syscall.SIGTERM)
-
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-
-	// opens a channel of all content created on the server, any errors detected,
-	// and scanning for the user input (Control + C)
-	go func() {
-		defer wg.Done()
-
-		// loop until user kills the channel input
-		for {
-			select {
-			case <-trapCh:
-				console.Println()
-				close(watchObj.doneChan)
-				return
-			case event, ok := <-watchObj.Events():
-				if !ok {
-					return
-				}
-
-				time, _ := time.Parse(time.RFC822, event.Time)
-				fileContent := contentMessage{
-					Key:  alias + strings.TrimPrefix(event.Path, clnt.GetURL().String()),
-					Time: time,
-					Size: event.Size,
-				}
-
-				// check to see if the newly create object matches the given params
-				match := doFindMatch(ctx, fileContent)
-				if match && ctx.String("exec") != "" {
-					doFindExec(ctx, fileContent)
-				} else if match && ctx.String("print") != "" {
-					doFindPrint(ctx, fileContent)
-				} else if match {
-					printMsg(findMessage{fileContent})
-				}
-
-			case err, ok := <-watchObj.Errors():
-				if !ok {
-					return
-				}
-				errorIf(err, "Unable to watch for events.")
-				return
-			}
-		}
-	}()
-
-	wg.Wait()
-}
-
-// doFindMatch matches whether fileContent matches appropriately with standard
+// matchFind matches whether fileContent matches appropriately with standard
 // "pattern matching" flags requested by the user, such as "name", "path", "regex" ..etc.
-func doFindMatch(ctx *cli.Context, fileContent contentMessage) (match bool) {
+func matchFind(ctx *findContext, fileContent contentMessage) (match bool) {
 	match = true
-	if !ctx.Bool("or") {
-		if ctx.String("ignore") != "" {
-			match = !pathMatch(ctx.String("ignore"), fileContent.Key)
-		}
-		return match
-	}
-	if ctx.String("name") != "" {
-		match = nameMatch(ctx.String("name"), fileContent.Key)
-	} else if ctx.String("path") != "" {
-		match = pathMatch(ctx.String("path"), fileContent.Key)
-	} else if ctx.String("regex") != "" {
-		match = regexMatch(ctx.String("regex"), fileContent.Key)
-	} else if ctx.String("older") != "" {
-		match = olderThanMatch(ctx.String("older"), fileContent.Time)
-	} else if ctx.String("newer") != "" {
-		match = newerThanMatch(ctx.String("newer"), fileContent.Time)
-	} else if ctx.String("larger") != "" {
-		match = largerSizeMatch(ctx.String("larger"), fileContent.Size)
-	} else if ctx.String("smaller") != "" {
-		match = smallerSizeMatch(ctx.String("smaller"), fileContent.Size)
+	if match && ctx.ignorePattern != "" {
```

Comment:
match here is unncessary

Reply:
It is for completeness.. otherwise logically it is hard.

Reply:
OK